private_sharing/api_views.py

```diff
@@ -226,7 +227,31 @@ def post(self, request, format=None):
         return Response("success")
 
 
-class ProjectFileDirectUploadView(ProjectFormBaseView):
+class SaveDataTypesMixin(object):
+    """
+    Mixin to do some final verification of datatypes to be associated with the datafile
+    and to do the actual association.
+    """
+
+    @property
+    def good_datatypes(self):
+        """
+        Verifies that the supplied datatypes are valid.  Returns boolean.
```

Comment:
Was there some reason this couldn't be done in the form?

Reply:
Because passing objects to the form requires much more complexity and this is three lines plus a doc string.